tools/lint-buffer-holder/main.go

```diff
@@ -0,0 +1,354 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+// Command lint-buffer-holder checks that types used as gRPC requests and
+// contain references to the gRPC buffer from which they were unmarshalled
+// embed the mimirpb.BufferHolder type to avoid use-after-release bugs.
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"go/token"
+	"go/types"
+	"iter"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/grafana/dskit/flagext"
+	"github.com/grafana/regexp"
+	"golang.org/x/tools/go/packages"
+)
+
+func main() {
+	args := must(flagext.ParseFlagsAndArguments(new(flag.FlagSet)))
+	new(linter).run(args...)
+	os.Exit(exitCode)
```

Comment:
[nit] Rather than using a global for `exitCode`, what if `run` returned the exit code?